### DISCUSSION TITLE
Differentiating non differentiable

### DISCUSSION THREAD

**Discussion by jakubMitura14** [jax-ml/jax#12866] [Q&A]
Hello I am in early stages of the model that has non differentiable part (I want to create the graph - discrete structure - from the output of the CNN - hence gradient can back flow from GNN to CNN) . I know that there are some methods of combinatorial optimazation that may help in making the non differentiable problem differentiable. Example of such tool is InferOpt.jl - Hovewer it is not compatible with Jax. I there anything in jax that helps in such usecases

**Comment by krzysztofrusek** [comment]
Hi, take a look at  #12460

**Reply by jakubMitura14** [reply]
Thanks! I will analyze it
